java/sage/Carny.java

```diff
@@ -1159,61 +1617,87 @@ private void stdProcessing()
     newWPMap.keySet().removeAll(watchedPotsToClear);
 
     if (Sage.DBG) System.out.println("CARNY Negative Energy Size: " + (blackBalled.size() + stoners.length));
-    for (Airing badAir : blackBalled)
+    for (int i = 0, blackBalledSize = blackBalled.size(); i < blackBalledSize; i++)
```

Comment:
Mainly curious why you undid the automatic iterator...are they actually slower? I figured they'd compile it to something highly optimal, but have never looked into it deeply.